### DISCUSSION TITLE
Matching multiple patterns

### DISCUSSION THREAD

**Discussion by aidalgol** [BurntSushi/ripgrep#2680] [Q&A] [answered]
I often want to get a list of all files in a source tree that match exactly two patterns.  I see that you can specify multiple patterns with `-e`, but that is an "OR" operation (it returns all files that match _any_ of the input patterns), whereas I'm after an "AND" operation (return only files that match _all_ of the input patterns.

I have been accomplishing this by chaining multiple `rg` invocations, but I wanted to find out whether this can be accomplished with a single `rg` invocation.

**Comment by BurntSushi** [comment, marked as answer]
> I have been accomplishing this by chaining multiple rg invocations

This is the standard solution.

Otherwise, see: https://github.com/BurntSushi/ripgrep/issues/875
